src/service/connect/items.ts

```diff
@@ -0,0 +1,44 @@
+import { OPConnect } from "@1password/connect";
+import type { SecretReference } from "../types";
+
+/**
+ * `op://<vault-name>/<item-name>/[section-name/]<field-name>`
+ */
+export const ref_regex =
+	/^op:\/\/(?<vault_name>[^/]+)\/(?<item_name>[^/]+)\/((?<section_name>[^/]+)\/)?(?<field_name>[^/]+)$/;
+
+export async function resolve_by_path(
+	client: OPConnect,
+	path: string,
+): Promise<string | undefined> {
+	const match = ref_regex.exec(path);
+	if (!match) {
+		throw new Error(`Invalid secret reference: ${path}`);
+	}
+	const { vault_name, item_name, section_name, field_name } =
+		match.groups as SecretReference;
+
+	const vault = await client.getVault(vault_name);
+	if (!vault.id) {
+		return undefined;
+	}
+	const item = await client.getItem(vault.id, item_name);
+
+	let item_fields = item.fields;
+	if (section_name) {
+		// how to deal with same label and id?
+		const section = item.sections?.filter(
+			(s) => s.label === section_name || s.id === section_name,
+		);
```

Comment:
solved by 847167768f217c284205706f5f858a595240e0cd , it will throw

- `The item does not have a field '{filed_name}'`
- `The item has more than one '{filed_name}' field`